core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2883,26 +2885,43 @@ private void registerCreateTableQuery(SqlValidatorScope parentScope,
       SqlCreateTable createTable,
       SqlNode enclosingNode,
       boolean forceNullable) {
-    //TODO: I'll likely need some sort of call to
+    // TODO: I'll likely need some sort of call to
     // validateFeature()
     // to confirm that the sql dialect we're validating for even supports CREATE_TABLE.
     // This will be done as followup: https://bodo.atlassian.net/browse/BE-4429
 
     final SqlNode queryNode = createTable.getQuery();
 
     // NOTE: query can be null, in the case that we're just doing a table definition with no data.
-    // For now, only supporting the case where we have a query
+    // For now, only supporting the case where we have a query, or a table identifier
     if (queryNode != null) {
-      registerQuery(
-          parentScope, //Should this be createTableNs?
-          usingScope, //Should be null?
-          queryNode,
+      if (!(queryNode instanceof SqlIdentifier)) {
+        //In the case that the query is a select statement, we need to register it and
+        // it's sub queries.
+        registerQuery(
+            parentScope,
+            usingScope,
+            queryNode,
+            enclosingNode,
+            null,
+            false
+        );
+      } else {
+        // Modified version of the code for registering the namespace of an
+        // identifier in registerFrom.
+        final SqlValidatorNamespace newNs;
```

Comment:
Just assign this on the same line you're creating it.

Reply:
Done